### PR TITLE
get_transformer returns pickle transformer if type is unsupported 

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Fixed https://flyte-org.slack.com/archives/CP2HDHKE1/p1686574457772509.
This pr also allows users to use pickle in the dataclass.

`get_transformer` returns a pickle transformer if the type is unsupported

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pingsutw** [comment]
cc @ggydush

**Review by wild-endeavor — DISMISSED** [review]

**Review by wild-endeavor — APPROVED** [review]
